internal/observers/kubernetes/api.go

```diff
@@ -277,21 +286,28 @@ func (o *Observer) endpointsInPod(pod *v1.Pod, client *k8s.Clientset, portAnnota
 			endpoint.PortType = services.PortType(port.Protocol)
 			endpoint.Port = uint16(port.ContainerPort)
 
-			container := &services.Container{
-				ID:      containerID,
-				Names:   []string{containerName},
-				Image:   container.Image,
-				Command: "",
-				State:   containerState,
-				Labels:  pod.Labels,
-			}
-
 			endpoint.AddExtraField("kubernetes_annotations", pod.Annotations)
 
 			endpoints = append(endpoints, &services.ContainerEndpoint{
 				EndpointCore:  *endpoint,
 				AltPort:       0,
-				Container:     *container,
+				Container:     *endpointContainer,
+				Orchestration: *orchestration,
+			})
+		}
+
+		if len(container.Ports) == 0 {
```

Comment:
Does this take into account the ports declared via annotations?

I like the concept of an endpoint being a whole container but I'm wary of how it will interact with some of the more indirect ways of defining ports (e.g. K8s annotations or Docker labels).

I kind of think we should _always_ emit a "container/portless" endpoint but I'm not sure how to make it backwards-compatible.

Reply:
@keitwb I looked into the annotations stuff more. I guess right now it's built around container port names. I don't see a clean way to extend the annotation naming convention that's there of `config.<port name>`. If we were doing it over maybe:

```
config.port.<name>
config.container.<name>
```

To keep backwards compat. might have to do `containerConfig.<name>` or something?

Reply:
Or maybe we could look for a port of that name and if it doesn't exist look for a container of that name. 🤷‍♂